swing/src/net/sf/openrocket/utils/SimulationTableToCSVFileExporter.java

```diff
@@ -0,0 +1,237 @@
+
+		// I'm pretty sure with the enablement/disablement of the menu item under the File dropdown,
+		// that this would no longer be needed because if there is no sim table yet, the context menu
+		// won't show up.   But I'm going to leave this in just in case....
+		if (modelRowCount <= 0) {
+			String msg = trans.get("simpanel.dlg.no.simulation.table.rows");
+			JOptionPane.showMessageDialog(tableParent, msg);
+			return;
+		}
+
+		JFileChooser fch = this.setUpFileChooser();
+		int selectionStatus = fch.showSaveDialog(tableParent);
+		if (selectionStatus == JFileChooser.CANCEL_OPTION || selectionStatus == JFileChooser.ERROR_OPTION) {
+			return;  // cancel or error... nothing to do here
+		}
+
+		fieldSep = ((CsvOptionPanel) fch.getAccessory()).getFieldSeparator();
+		int precision = ((CsvOptionPanel) fch.getAccessory()).getDecimalPlaces();
+		((CsvOptionPanel) fch.getAccessory()).storePreferences();
+
+		if (fieldSep.equals(SPACE)) {
+			fieldSep = " ";
+		} else if (fieldSep.equals(TAB)) {
+			fieldSep = "\t";
+		}
+
+		File csvFile = fch.getSelectedFile();
+
+		String csvSimResultString = "";
+		// obtain the column titles for the first row of the csv
+		ArrayList<String> rowColumnElement = new ArrayList<>();
+		for (int j=1; j<modelColumnCount ; j++) {
+			String colName = simulationTable.getColumnName(j);
+			String unitString = valueColumnToUnitString.get(colName);
+			if (unitString != null) {
+				colName += " (" + unitString + ")";
+			}
+			rowColumnElement.add(colName);
+		}
+
+		// ONE difference here is that we'll place any warnings at the last cell in the csv.
+		csvSimResultString = StringUtils.join(fieldSep, rowColumnElement) + fieldSep + "Simulation Warnings";
+
+		String fullOutputResult = csvSimResultString;
+		
+		// get relevant data and create the comma separated data from it.
+		for (int i1 = 0; i1 < modelRowCount; i1++) {
+			// account for sorting... resulting csv file will be in the
+			// same order as shown in the table thanks to this gem.
+			int i = simulationTable.convertRowIndexToModel(i1);
+
+			int nullCnt = 0;
+			rowColumnElement.clear();
+
+			// get the simulation's warning text if any... this bypasses the need to use
+			// the column 0 stuff which is kind of difficult to use!
+			WarningSet ws = document.getSimulation(i).getSimulatedWarnings();
+			String warningsText = "";
+			for (Warning w : ws) {
+				String warning = w.toString();
+				if (warning != null) {
+					warningsText += w + " "; // TODO - formatting.  inserting a \n does funny things so use " " for now
+				}
+			}
+
+			// piece together the column data for the ith row, skipping any rows with null counts > 0!
+			for (int j=1; j<modelColumnCount ; j++) { // skip first column
+				String colName = simulationTable.getColumnName(j);
+				String unitString = valueColumnToUnitString.get(colName); // unit string MAY be null!
+
+				Object o = simulationTableModel.getValueAt(i, j);
+				if (o != null) {
+					String value = null;
```

Comment:
Make this a final String and remove the null initialization, it's better practice since that way you'll be sure that value will never be actually null. (take a look at my initial remark on this and the code provided)

Reply:
done